synapse/cortex.py

```diff
@@ -865,33 +901,53 @@ def getCoreMods(self):
             ret.append((modname, mod.conf))
         return ret
 
-    def getForkView(self, iden):
-        pass
+    def eval(self, text, opts=None, user=None):
+        with self.snap(user=user) as snap:
+            yield from snap.eval(text, opts=opts, user=user)
 
-    def newForkView(self):
-        pass
+    def storm(self, text, opts=None, user=None):
 
-    def eval(self, text, opts=None):
-        '''
-        Evaluate a storm query and yield Nodes only.
-        '''
-        query = self.view.getStormQuery(text, opts=opts)
-        for node, path in query.evaluate():
-            yield node
+        chan = s_queue.Queue()
+        query = self.getStormQuery(text)
 
-    def storm(self, text, opts=None):
-        '''
-        Evaluate a storm query and yield result messages.
+        self._runStormThread(query, opts=opts, user=user, chan=chan)
 
-        Args:
-            text (str): A storm query.
+        for mesg in chan:
+            yield mesg
 
-        Yields:
-            ((str,dict)): Storm messages.
+    @s_glob.inpool
+    def _runStormThread(self, query, opts=None, user=None, chan=None):
+
+        try:
+
+            count = 0
+            tick = s_common.now()
```

Comment:
personal nit - i'd move count and tick outside of the try: block to make variable initialization used by finally: a pretty sure thing (if `now()` fails we have bigger issues)

Reply:
:+1: